### ISSUE TITLE
How related is this project to the GeminiGraph project?

### ISSUE THREAD

**Issue by winterland1989** [Tencent/plato#8]
I guess you should not just cite the Gemini paper for "inspiration and guide" in the README.md when you literally lifting and shifting codes from [the author's git repo](https://github.com/thu-pacman/GeminiGraph).

I cannot help feeling this looks a bit like another [go-chassis vs go-micro scandal]( https://github.com/go-chassis/go-chassis/issues/28), only this time it was from Tencent.

**Comment by winterland1989** [comment]
d CSDN, 
how hard is it to give projects like GeminiGraph a proper acknowledgement they deserve?
Never assume originality by dodging necessary acknowledgement.

**Comment by yzhwang** [comment]
Hi @winterland1989, thanks for your interest in Plato project. I will address the issue in representative for the whole dev team of Plato.

First and foremost, Plato did refer to and has integrate several state-of-the-art design principles from both academia and industry, including Gemini. To show our acknowledgement to the Gemini team, we have included the reference to Gemini's paper in our original open source version:
https://github.com/Tencent/plato/blob/8cb33dd121c90e86ebe8a677592e4992a1062e28/plato/graph/structure/bcsr.hpp#L61
https://github.com/Tencent/plato/blob/8cb33dd121c90e86ebe8a677592e4992a1062e28/plato/graph/partition/sequence.hpp#L135
However, we did not put acknowledgement to Gemini in our project page and for the two files contain utility functions (wrapper of atomic primitives and MPI data types) we did not add LICENCE information accordingly.

These are clearly oversight and we realized internally soon after the open source. To fix this, we made response immediately (on Nov 15th 2019, GTM+8). First we added proper LICENCE info. Then we put acknowledgement to Gemini and other teams on project's front page to make it more rigorous and truly reflect Gemini's influence to our work. We have also reached out to the original Gemini team, and us two parties have made agreement to the current fix with the recognition of the original Gemini team. Currently the Gemini part reads as follows:
**Special thanks to Xiaowei Zhu and many for their work Gemini. Several basic utility functions in Plato is derived from Gemini, the design principle of some dual-mode based algorithms in Plato is also heavily influenced by Gemini's dualmode-engine.**
We will make proper acknowledgement to Gemini team and all other teams in all our external publicity from now on. In summary, we will draw lessons from this issue and treat open source-related issues more carefully and thoroughly.

We hereby thanks @winterland1989 again for pointing out this issue. We sincerely hope Plato could grow with the help from the open source community and hope our little contribution to graph computing framework could benefit more researchers and developers in the field.
